### PR TITLE
chore: add kind support for 1.25

### PR DESCRIPTION
**Pull request by hectorj2f**
#### Summary

We can now test Kubernetes v1.25. This PR adds the changes to actions so it uses v1.25.

https://github.com/kubernetes-sigs/kind/releases

### CODE REVIEW

**Comment by hectorj2f** [comment]
related to https://github.com/sigstore/scaffolding/pull/343

**Comment by hectorj2f** [comment]
We are blocked by https://github.com/sigstore/scaffolding/pull/343, we need to use the new scaffolding version when available.

```
 ./setup-kind.sh --registry-url registry.local:5000 --cluster-suffix cluster.local --k8s-version v1.25.x --knative-version 1.6.0
RUNNING ON Linux
Unsupported version: v1.25.x
```

**Comment by hectorj2f** [comment]
This PR is blocked until a new scaffolding version gets released. 

```
+ echo 'Installing kind and knative using release'
+ curl -fLo ./setup-kind.sh https://github.com/sigstore/scaffolding/releases/download/v0.4.6/setup-kind.sh
```

**Comment by hectorj2f** [comment]
scaffolding 0.4.8 was released 3d ago, we could now use that version in our tests.

**Comment by hectorj2f** [comment]
@vaikas  @cpanato  This is ready now! 🎉 💃🏻

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -43,7 +43,7 @@ jobs:
           check-latest: true
 
       # will use the latest release available for ko
-      - uses: imjasonh/setup-ko@ace48d793556083a76f1e3e6068850c1f4a369aa # v0.6
+      - uses: imjasonh/setup-ko@v0.6 # v0.6
```
Comment (cpanato) [inline .github/workflows/build.yaml:46]: please use the git hash instead of the tag

File: .github/workflows/kind-cluster-image-policy.yaml [hunk 2]
```diff
@@ -57,8 +58,8 @@ jobs:
         go-version: '1.18'
         check-latest: true
 
-    # will use the latest release available for ko
-    - uses: imjasonh/setup-ko@ace48d793556083a76f1e3e6068850c1f4a369aa # v0.6
+    # will use the latest release akindest/nodevailable for ko
+    - uses: imjasonh/setup-ko@v0.6 # v0.6
```
Comment (cpanato) [inline .github/workflows/kind-cluster-image-policy.yaml:62]: same here

File: .github/workflows/kind-e2e-cosigned.yaml [hunk 3]
```diff
@@ -49,7 +50,7 @@ jobs:
         go-version: '1.18'
         check-latest: true
 
-    - uses: imjasonh/setup-ko@ace48d793556083a76f1e3e6068850c1f4a369aa # v0.6
+    - uses: imjasonh/setup-ko@v0.6 # v0.6
```
Comment (cpanato) [inline .github/workflows/kind-e2e-cosigned.yaml:53]: and here :)

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -27,7 +27,7 @@ jobs:
 
       - uses: anchore/sbom-action/download-syft@b5042e9d19d8b32849779bfe17673ff84aec702d # v0.11.0
 
-      - uses: imjasonh/setup-ko@ace48d793556083a76f1e3e6068850c1f4a369aa # v0.6
+      - uses: imjasonh/setup-ko@v0.6 # v0.6
```
Comment (cpanato) [inline .github/workflows/release.yaml:30]: also here :)

**Review by cpanato — CHANGES_REQUESTED** [review]
small change request

**Review by vaikas — APPROVED** [review]
Thanks!

**Review by cpanato — APPROVED** [review]
thank you

**Pull request merged**